File: 백준/Silver/17413. 단어 뒤집기 2/단어 뒤집기 2.swift
while let input = readLine(){
    let str = input.map{String($0)}
    var ans = ""
    var ori = ""
    var check = 0
    var stack = [String]()
    for i in 0...str.count-1{
        if(str[i] == "<"){
            if(!stack.isEmpty){
                for i in stride(from: stack.count-1, to: -1, by: -1){
                    ans += stack[i]
                }
                stack.removeAll()
            }
            ori += str[i]
            check = 1
        }
        else if(str[i] == ">"){
            ori += str[i]
            ans += ori
            ori = ""
            check = 0
        }
        else{
            if(check == 1){
                ori += str[i]
            }
            else{
                if(str[i] == " "){
                    for i in stride(from: stack.count-1, to: -1, by: -1){
                        ans += stack[i]
                    }
                    ans += " "
                    stack.removeAll()
                }
                else{
                    stack.append(str[i])
                }
                if(i == str.count-1){
                    for i in stride(from: stack.count-1, to: -1, by: -1){
                        ans += stack[i]
                    }
                }
            }
        }
    }
    print(ans)
}

File: 백준/Silver/17413. 단어 뒤집기 2/단어 뒤집기 2.java
import java.io.*;
import java.lang.reflect.Array;
import java.util.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        String [] A = br.readLine().split("");
        Stack<String>stack1= new Stack<>();
        Stack<String>stack2= new Stack<>();
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < A.length; i++) {
            if(A[i].equals("<")){
                if(!stack1.isEmpty()){
                    while (!stack1.isEmpty()) {
                        sb.append(stack1.pop());
                    }
                }
                while (!A[i].equals(">")) {
                    sb.append(A[i]);
                    i++;
                }
                sb.append(">");
            }
            else {
                if (!A[i].equals(" ")) {
                    stack1.push(A[i]);
                }
                if (!stack1.isEmpty() && A[i].equals("<")) {
                    while (!stack1.isEmpty()) {
                        sb.append(stack1.pop());
                    }
                }
                if (i == A.length - 1 || A[i].equals(" ")) {
                    while (!stack1.isEmpty()) {
                        sb.append(stack1.pop());
                    }
                    sb.append(" ");
                }
            }
        }
        System.out.println(sb);
        bw.flush();
        bw.close();
        br.close();
    }

}